### PR TITLE
CR-1089460 Add changes to show heartbeat status from xmc through "xbutil examine" command

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Display heartbeat status from xmc subdev using xbutil2 examine command

Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
Below is the output observed on U200 setup

```
# xbutil examine -r xmc

-----------------------------------------------------
1/1 [0000:86:00.1] : xilinx_u200_gen3x16_xdma_base_1
-----------------------------------------------------
XMC status
  heartbeat_stall        : false
  heartbeat_count        : 7320
  heartbeat_err_code     : 0x100
  heartbeat_err_time     : 0

```

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/tools/common/ReportXmc.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    pt.put("heartbeat_count", xrt_core::device_query<xrt_core::query::heartbeat_count>(_pDevice));
+    pt.put("heartbeat_err_code", xrt_core::device_query<xrt_core::query::heartbeat_err_code>(_pDevice));
+    pt.put("heartbeat_stall", xrt_core::device_query<xrt_core::query::heartbeat_stall>(_pDevice));
+  } catch(...) {}
+  // There can only be 1 root node
+  _pt.add_child("xmc", pt);
+}
+
+void
+ReportXmc::writeReport( const xrt_core::device* /*_pDevice*/,
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportXmc.cpp:48]: Please change the name to `ReportXmcStatus`

File: src/runtime_src/core/tools/common/ReportXmc.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                                           boost::property_tree::ptree &_pt) const
+{
+  boost::property_tree::ptree pt;
+  try {
+    pt.put("Description", "XMC status");
+    pt.put("heartbeat_err_time", xrt_core::device_query<xrt_core::query::heartbeat_err_time>(_pDevice));
+    pt.put("heartbeat_count", xrt_core::device_query<xrt_core::query::heartbeat_count>(_pDevice));
+    pt.put("heartbeat_err_code", xrt_core::device_query<xrt_core::query::heartbeat_err_code>(_pDevice));
+    pt.put("heartbeat_stall", xrt_core::device_query<xrt_core::query::heartbeat_stall>(_pDevice));
+  } catch(...) {}
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportXmc.cpp:42]: Never do this kind of catch. @stsoe has created a new exception `xrt_core::query::error` which catches errors related to query requests (no_such_key, sysfs_error, a few more), but does not catch unrelated "real" errors.

File: src/runtime_src/core/tools/common/ReportXmc.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ReportXmc::getPropertyTree20202( const xrt_core::device * _pDevice,
+                                           boost::property_tree::ptree &_pt) const
+{
+  boost::property_tree::ptree pt;
+  try {
+    pt.put("Description", "XMC status");
+    pt.put("heartbeat_err_time", xrt_core::device_query<xrt_core::query::heartbeat_err_time>(_pDevice));
+    pt.put("heartbeat_count", xrt_core::device_query<xrt_core::query::heartbeat_count>(_pDevice));
+    pt.put("heartbeat_err_code", xrt_core::device_query<xrt_core::query::heartbeat_err_code>(_pDevice));
+    pt.put("heartbeat_stall", xrt_core::device_query<xrt_core::query::heartbeat_stall>(_pDevice));
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportXmc.cpp:41]: Maybe rename "heatbeat_stall" to something more human-readable... I am not sure what is a more suitable tag

Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportXmc.cpp:41]: Maybe rename "heatbeat_stall" to something more human-readable... I am not sure what is a more suitable tag

**Review by AShivangi — CHANGES_REQUESTED** [review]
Please address the requested changes and add the json output to the PR desc

**Review by AShivangi — APPROVED** [review]
This looks good! Thanks for enhancing the output

**Pull request merged**